11-RegEx/regex-url.js

```diff
@@ -0,0 +1,7 @@
+function ExtractUrl(string){
+var REGEX= /(?:(?<Host>[^\:]*)\:\/\/)?(?:(?<Username>[^\:\@]*)(?:\:([^\@]*))?\@)?(?:(?<Domain>[^\/\:]*))?(?:\:(?<Port>[0-9]*))?(?<Path>\/[^\?#.]*(?=.*?\/)\/)?((?<Filename>[^\?#]*)(?<Extension>\w*[.]\w{3,4}))?(?:\?(?<Query>[^#]*))?(?:#(.*))?/
```

Comment:
Almost perfect, but can you think of extraction of the elements only in case the string is URL (e.g. ExtractUrl('ftp://') would return the protocol but this is not a URL)

Reply:
If I am not mistaken I should do that using js functionalities such as "if" statements and not to touch the regex .